### DISCUSSION TITLE
Can you help transplant the MicroPython firmware of ESP32P4?

### DISCUSSION THREAD

**Discussion by Vincent1-python** [micropython/micropython#15826] [ESP32]
### Description

Can you help transplant the MicroPython firmware of ESP32P4?

### Code Size

Can you help transplant the MicroPython firmware of ESP32P4?

### Implementation

I hope the MicroPython maintainers or community will implement this feature, I intend to implement this feature and would submit a Pull Request if desirable, I would like to sponsor development of this feature

### Code of Conduct

Yes, I agree

**Comment by bidrohini** [comment]
Support for ESP32-P4 should be added.

**Comment by projectgus** [comment]
I have seen comments suggesting @mattytrentini is looking into this, but there's no PR for pending support yet.

**Reply by mattytrentini** [reply]
I have made a basic start - at the prompting of @urish because he was interested in adding the P4 to Wokwi even before the device was released! But I ran into too many issues at the time since the IDF was under _heavy_ development. 

I recently received a P4 dev board and am hoping to take another look at this soon but I'm not sure when I'll get back to it. 

@tannewt has recently raised a [PR for CircuitPython](https://github.com/adafruit/circuitpython/pull/9612) and @bwhitman has dabbled with a [P4 port for CircuitPython too](https://github.com/adafruit/circuitpython/compare/main...bwhitman:circuitpython:main). Both of those should help resolve issues that a MicroPython port may run into.

**Reply by tannewt** [reply]
Upstream TinyUSB doesn't have P4 support yet. It is being tracked here: https://github.com/hathach/tinyusb/issues/2791

Note my PR is based on @bwhitman's code. I just got it fixed up so it'd run. They did most of the grunt work for it.

**Reply by Vincent1-python** [reply]
Are you ready？

**Comment by Lzw655** [comment]
Hi @mattytrentini, thank you very much for your work. May I ask if you have made any new progress?

**Comment by cheops** [comment]
@mattytrentini  is some of your (pre-alpha?) code somewhere available, so that others might pick it up?
esp32p4 is already available in devkit and some boards with displays are appearing online.

**Reply by mattytrentini** [reply]
I tried again a couple of weeks ago and ran into compilation errors (which appeared to be in IDF code, so I must have set up something incorrectly). It's not online anywhere, and it's in a very hack-y state but I'll try and find time to have another go at it, tidy it up and push it up somewhere.

**Reply by cheops** [reply]
Hello @mattytrentini
any progress on the micropython port for esp32 p4?

Thanks

**Comment by nubcore** [comment]
I have a large project underway with a particular interest in this. I've seen Circuit Python has done some early builds with it.

Very much looking forward to seeing what @mattytrentini has been able to do.

Does anyone have particular experience or thoughts on dev kits, or any of the hand full of boards floating around, particularly if they have C6 as well?

I am setting up a new build environment and will hopefully be able to offer some testing soon.

**Comment by lbuque** [comment]
https://github.com/m5stack/uiflow-micropython/tree/master/m5stack/boards/M5STACK_Tab5

**Comment by jonnor** [comment]
There is an initial MR for ESP32P4 now, at https://github.com/micropython/micropython/pull/17356
